src/microbe_stage/organelle_components/PilusComponent.cs

```diff
@@ -33,41 +42,40 @@ protected override bool NeedsUpdateAnyway()
             membraneCoords *= 0.5f;
         }
 
-        float pilusSize = 4.6f;
-
-        // Scale the size down for bacteria
-        if (organelle.ParentMicrobe.Species.IsBacteria)
-        {
-            pilusSize *= 0.5f;
-        }
```

Comment:
I don't fully get why moving this code to a new method that is in CustomAttach works. It should be anyway so that OnPositionChanged **should** only be called when truly needed. Fixing OnPositionChanged being called incorrectly will likely also fix things for the flagella being repositioned all the time.

Reply:
The way I understood the code previously, OnPositionChanged happens every frame the organelle moves. This means every single frame the player moved, DestroyShape() was being called, and a new shape was being created.

This new way, when the organelle is created the CollisonShapes are also created. OnPositionChanged only changes the transform of these shapes now, as you would expect.

Reply:
`OnPositionChanged` **should** only be called when the position within the cell relative coordinates for the thing changes. So basically whenever an organelle is added or removed and the membrane is regenerated, the membrane point the pilus or flagellum wants to attach can change, and **then** OnPositionChanged is called. 
It's a good change to only recreate the shape when needed, but it should also take into account that if the pilus size changes the physics shape needs editing.

Reply:
>It's a good change to only recreate the shape when needed, but it should also take into account that if the pilus size changes the physics shape needs editing.

This can be done here, we just add scaling to the transform. I had it doing that before I took out scaling for the Pilus.

I will check how often OnPositionChanged is called, I was misinterpreting its function.

Reply:
Yeah it's definitely called every single frame. I'll take a look...